src/core_plugins/kibana/public/context/query_parameters/actions.js

```diff
@@ -43,7 +46,13 @@ export function QueryParameterActionsProvider() {
     )
   );
 
+  const addTermsFilter = (state) => (field, values, operation) => {
+    const { queryParameters: { indexPatternId } } = state;
```

Comment:
Does this provide a benefit over `const indexPatternId = state.queryParameters.indexPatternId`? Does it not throw if `queryParameters` is undefined or something? Just trying to understand the preference for destructing here since a regular assignment seems much more readable to me.